be2-scala/src/test/scala/ch/epfl/pop/storage/DbActorNewSuite.scala

```diff
@@ -274,4 +274,159 @@ class DbActorNewSuite extends TestKit(ActorSystem("DbActorNewSuiteActorSystem"))
+    val channelData: ChannelData = ChannelData(ObjectType.LAO, Nil)
+    initialStorage.write((CHANNEL_NAME, channelData.toJsonString))
+
+    val dbActor: AskableActorRef = system.actorOf(Props(DbActorNew(mediatorRef, initialStorage)))
+
+    val ask = dbActor ? DbActorNew.ReadChannelData(channelName1)
+    val answer = Await.result(ask, duration)
+
+    answer shouldBe a[DbActorNew.DbActorReadChannelDataAck]
+
+    val readChannelData: ChannelData = answer.asInstanceOf[DbActorNew.DbActorReadChannelDataAck].channelData
+
+    readChannelData should equal(channelData)
+  }
+
+  test("catchup works on a channel with valid ChannelData and messages"){
+    val channelName1: Channel = Channel(CHANNEL_NAME)
+    val initialStorage: InMemoryStorage = InMemoryStorage()
+    val message2 = MESSAGE.copy(message_id = Hash(Base64Data("RmFrZSBtZXNzYWdlX2lkIDopIE5vIGVhc3RlciBlZ2cgcmlnaHQgdGhlcmUhIC0tIE5pY29sYXMgUmF1bGlu")))
+    val listIds: List[Hash] = MESSAGE.message_id :: message2.message_id :: Nil
+    val channelData: ChannelData = ChannelData(ObjectType.LAO, listIds)
+    initialStorage.write(
+      (CHANNEL_NAME, channelData.toJsonString), 
+      (s"$CHANNEL_NAME${Channel.SPECIAL_SEPARATOR}${MESSAGE.message_id}", MESSAGE.toJsonString),
+      (s"$CHANNEL_NAME${Channel.SPECIAL_SEPARATOR}${message2.message_id}", message2.toJsonString),
+    )
+
+    val dbActor: AskableActorRef = system.actorOf(Props(DbActorNew(mediatorRef, initialStorage)))
+
+    val ask = dbActor ? DbActorNew.Catchup(channelName1)
+    val answer = Await.result(ask, duration)
+
+    answer shouldBe a[DbActorNew.DbActorCatchupAck]
+
+    val list: List[Message] = answer.asInstanceOf[DbActorNew.DbActorCatchupAck].messages
+
+    list.size should equal(2)
+    list should contain(MESSAGE)
+    list should contain(message2)
+
+  }
+
+  test("catchup should not fail on a channel with ChannelData containing missing message_ids (and only return valid messages)"){
+    val channelName1: Channel = Channel(CHANNEL_NAME)
+    val initialStorage: InMemoryStorage = InMemoryStorage()
+    val message2 = MESSAGE.copy(message_id = Hash(Base64Data("RmFrZSBtZXNzYWdlX2lkIDopIE5vIGVhc3RlciBlZ2cgcmlnaHQgdGhlcmUhIC0tIE5pY29sYXMgUmF1bGlu")))
+    val listIds: List[Hash] = MESSAGE.message_id :: message2.message_id :: Nil
+    val channelData: ChannelData = ChannelData(ObjectType.LAO, listIds)
+    initialStorage.write(
+      (CHANNEL_NAME, channelData.toJsonString), 
+      (s"$CHANNEL_NAME${Channel.SPECIAL_SEPARATOR}${MESSAGE.message_id}", MESSAGE.toJsonString),
+    )
+
+    val dbActor: AskableActorRef = system.actorOf(Props(DbActorNew(mediatorRef, initialStorage)))
+
+    val ask = dbActor ? DbActorNew.Catchup(channelName1)
+    val answer = Await.result(ask, duration)
+
+    answer shouldBe a[DbActorNew.DbActorCatchupAck]
+
+    val list: List[Message] = answer.asInstanceOf[DbActorNew.DbActorCatchupAck].messages
+
+    list.size should equal(1)
+    list should contain(MESSAGE)
+
```

Comment:
sometimes when you need to write a rather complex test, it's convenient for the readers to clearly structure the test as the 3 As (arrange, act, assert), using comments to highlight/summarize the structure. Here the test title is clear enough, but as an example of what could be "bare minimum" comments to ease the reading of the test:
```
    // arrange
    val channelName1: Channel = Channel(CHANNEL_NAME)
    ...
    val dbActor: AskableActorRef = system.actorOf(Props(DbActorNew(mediatorRef, initialStorage)))

    // act
    val ask = dbActor ? DbActorNew.Catchup(channelName1)
    val answer = Await.result(ask, duration)

    // assert
    answer shouldBe a[DbActorNew.DbActorCatchupAck]
```